.github/workflows/clang-format.yml

```diff
@@ -0,0 +1,40 @@
+name: "Check clang format"
```

Comment:
What about adding all of this directly into the `linter.yml`? That workflow is quite fast and I think it belongs there.